packages/legacy/src/queries/hosts.ts

```diff
@@ -80,30 +80,37 @@ const generateSecret = (
   host: IHost,
   provider: IVMwareProvider,
   hostConfig?: IHostConfig
-): ISecret => ({
-  apiVersion: 'v1',
-  data: {
-    user: values.adminUsername && btoa(values.adminUsername),
-    password: values.adminPassword && btoa(values.adminPassword),
-  },
-  kind: 'Secret',
-  metadata: {
-    ...(secretBeingReusedRef || {
-      generateName: truncateK8sString(provider.name, `-${host.id}-`, true),
-      namespace: provider.namespace,
-    }),
-    labels: {
-      createdForResourceType: ForkliftResourceKind.Host,
-      createdForResource: getHostConfigRef(provider, host).name,
+): ISecret => {
+  const matchingNetworkAdapter = host.networkAdapters.find(
+    ({ name }) => values.selectedNetworkAdapter?.name === name
+  );
+  return {
+    apiVersion: 'v1',
+    data: {
+      user: values.adminUsername && btoa(values.adminUsername),
+      password: values.adminPassword && btoa(values.adminPassword),
+      provider: 'vsphere',
```

Comment:
can you add the `provider` and `ip` fields to the `ISecret`  struct ?

https://github.com/kubev2v/forklift-console-
plugin/blob/main/packages/legacy/src/queries/types/secrets.types.ts#L4

should look somthing like:
``` ts
provider?: string;
ip?: string;
```